producer-services/src/main/scala/za/co/absa/spline/producer/service/repo/RecursiveSchemaFinder.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline.producer.service.repo
+
+import za.co.absa.spline.producer.model.OperationLike
+
+import scala.collection.mutable
+
+class RecursiveSchemaFinder(operations: Seq[OperationLike]) {
+  private val schemaByOperationIdCollector = mutable.Map.empty[Int, Option[Any]]
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [It's preferred to use immutable collections instead of mutable ones](https://app.codacy.com/app/ABSA_OSS/spline/pullRequest?prid=4594774)